### PR TITLE
updated priors vignette so that it works, exported required functions

### PR DESCRIPTION
**Pull request by dlebauer**
Updated the prior vignette so that it can compile. This required exporting a few functions from the PEcAn.priors function. Or perhaps these were incorrectly identified as S3 methods.

## Motivation and Context

To get the vignette working again.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [x] When possible
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ For more information about this file see also [Keep a Changelog](http://keepacha
 - Replace deprecated `rlang::UQ` syntax with the recommended `!!`
 - Explicitly use `PEcAn.uncertainty::read.ensemble.output` in `PEcAn.utils::get.results`. Otherwise, it would sometimes use the deprecated `PEcAn.utils::read.ensemble.output` version.
 - History page would not pass the hostname parameter when showing a running workflow, this would result in the running page showing an error.
+- Fixed Priors vignette
```
Comment (infotroph) [inline CHANGELOG.md]:
```suggestion
- Fixed Priors vignette (#2439)
```

File: modules/priors/vignettes/priors_demo.Rmd [hunk 2]
```diff
@@ -160,13 +176,14 @@ wullschleger.data <- data.frame(Y  = wullschleger.data$corrected.vcmax,
...
                                 n  = wullschleger.data$neff)
 ## Summarize data by species
-wullschleger.vcmax <- ddply(wullschleger.data, 
-                            .(sp),
-                            summarise, 
-                            Y = mean(Y), 
-                            obs.prec = 1/sqrt(sum((1/obs.prec)^2)), 
-                            ft = mean(ft), # identity
-                            n =  sum(n)) [,-1]
+library(dplyr)
```
Comment (infotroph) [inline modules/priors/vignettes/priors_demo.Rmd]: already loaded dplyr at the top of the file — is this call redundant or is it needed for some  Rmarkdown per-chunk subtlety here?

Comment (dlebauer) [inline modules/priors/vignettes/priors_demo.Rmd]:
```suggestion

```

Comment (dlebauer) [reply modules/priors/vignettes/priors_demo.Rmd]: fixed in https://github.com/PecanProject/pecan/pull/2439/files/272fbc199168822d2486425a64e9236b3a697617

File: modules/priors/NAMESPACE [hunk 3]
```diff
@@ -1,15 +1,16 @@
 # Generated by roxygen2: do not edit by hand
 
-S3method(plot,posterior.density)
-S3method(plot,prior.density)
-S3method(plot,trait)
```
Comment (infotroph) [inline modules/priors/NAMESPACE]: @dlebauer Tried running this locally and my version of Roxygen wants to re-add these S3 method declarations. How did you persuade it to treat these as regular functions?

**Comment by dlebauer** [comment]
No idea! I think I just used devtools::document()

> --

_______________________

*David LeBauer*
Director of Data Sciences
Arizona Experiment Station
THE UNIVERSITY OF ARIZONA

Bioscience Research Labs, 207
1230 N Cherry Ave | Tucson, AZ 85721
Office: 520-621-4381
dlebauer@email.arizona.edu
orcid <https://orcid.org/0000-0001-7228-053X> | twitter
<https://twitter.com/dlebauer> | github <https://github.com/dlebauer> |
linkedin <https://www.linkedin.com/in/dlebauer/>

**Comment by infotroph** [comment]
>> my version of Roxygen wants to re-add these S3 method declarations. How did you persuade it to treat these as regular functions?

> No idea! I think I just used devtools::document()

@dlebauer Huh, weird. Well, I'd say fix the missing MASS declaration and see if the build passes -- if it does, the problem was on my end. If it complains about files changing during the build, then we'll need to revisit this

**Comment by infotroph** [comment]
@dlebauer Can you estimate when you'll have time to finish this? I'd like to use the priors package as a test case for #2437, but it'd be preferable to have this merged first to avoid conflict / take advantage of your work

**Comment by dlebauer** [comment]
@infotroph I've added MASS to imports

regarding the dependencies ... I think fitdistrplus, rjags, and rriskDistributions are critical to the vignette - since the vignette shows how to use these packages to come up with priors. DEoptim could possibly be replaced with optim and dplyr could certainly be replaced. But then this brings up the question of whether it is worthwhile to refactor code just to reduce dependencies (especially in vignettes, or in the case of dplyr, where the package is installed elsewhere in the PEcAn build).

File: modules/priors/R/plots.R [hunk 4]
```diff
@@ -191,38 +192,39 @@ plot.trait <- function(trait,
           axis.ticks.y = element_blank(), 
           axis.line.y = element_blank(), 
           legend.position = "none", 
-          plot.title = element_text(size = fontsize$title), 
+          plot_title = element_text(size = fontsize$title),
```
Comment (infotroph) [inline modules/priors/R/plots.R]: This one's a ggplot theme object and should stay dotted

File: modules/priors/R/plots.R [hunk 5]
```diff
@@ -78,7 +79,7 @@ priorfig <- function(priordata = NA, priordensity = NA, trait = "", xlim = "auto
           axis.text.x = element_text(size = fontsize),
           axis.title.y = element_blank(),   ## hide y axis label
           axis.title.x = element_text(size = fontsize * 0.9), 
-          plot.title = element_text(size = fontsize * 1.1))
+          plot_title = element_text(size = fontsize * 1.1))
```
Comment (dlebauer) [inline modules/priors/R/plots.R]:
```suggestion
          plot.title = element_text(size = fontsize * 1.1))
```

File: modules/priors/R/plots.R (hunk 4 above)
Comment (dlebauer) [reply modules/priors/R/plots.R]:
```suggestion
          plot.title = element_text(size = fontsize$title), 
```

**Comment by dlebauer** [comment]
Thanks for your help @infotroph! Finally can merge!

**Pull request merged**